### PR TITLE
N°6956 - Improve \CollectionPlan PHPDoc to ease DX in IDE

### PR DESCRIPTION
**Pull request by Hipska**
In order to let IDE get better instructions

### CODE REVIEW

**Review by piRGoif — APPROVED** [review]

File: core/collectionplan.class.inc.php [hunk 1]
```diff
@@ -40,17 +40,17 @@ public function Init(): void
 	}
 
 	/**
-	 * @return \CollectionPlan
+	 * @return static
 	 */
-	public static function GetPlan(): CollectionPlan
+	public static function GetPlan(): static
```
Comment (Molkobain) [inline core/collectionplan.class.inc.php:45]: This is not supported by PHP 7.x, only PHP 8.0+

One solution could be to remove the return type hint completely and keep the PHPDoc as you suggested. That's what we did in iTop 3.0 for UIBlock for example.

**Review by Molkobain — CHANGES_REQUESTED** [review]

File: core/collectionplan.class.inc.php (hunk 1 above)
Comment (Hipska) [reply core/collectionplan.class.inc.php:45]: That's right, will change to this:
```suggestion
	public static function GetPlan()
```

**Review by Molkobain — APPROVED** [review]

File: core/collectionplan.class.inc.php (hunk 1 above)
Comment (piRGoif) [reply core/collectionplan.class.inc.php]: Woops, good to know, good catch Guillaume, thanks !

**Comment by Molkobain** [comment]
Accepted during technical review

**Comment by Molkobain** [comment]
Accepted during functional review

**Pull request merged**